src/main/java/com/gluonhq/substrate/SubstrateDispatcher.java

```diff
@@ -67,25 +65,21 @@ public static void main(String[] args) throws Exception {
         Triplet targetTriplet = targetProfile != null? new Triplet(Constants.Profile.valueOf(targetProfile.toUpperCase()))
                 :Triplet.fromCurrentOS();
 
-        ProjectConfiguration config = new ProjectConfiguration();
-        config.setGraalPath(graalVM);
-        config.setMainClassName(mainClass);
-        config.setAppName(appName);
-        config.setJavafxStaticSdkVersion(Constants.DEFAULT_JAVAFX_STATIC_SDK_VERSION);
-        config.setTarget(targetTriplet);
+        ProjectConfiguration publicConfig = new ProjectConfiguration(mainClass);
+        publicConfig.setGraalPath(graalVM);
+        publicConfig.setAppName(appName);
+        publicConfig.setJavafxStaticSdkVersion(Constants.DEFAULT_JAVAFX_STATIC_SDK_VERSION);
+        publicConfig.setTarget(targetTriplet);
+
+        PrivateProjectConfiguration config = new PrivateProjectConfiguration(publicConfig);
+
         config.setUsePrismSW(usePrismSW);
         config.getIosSigningConfiguration().setSkipSigning(skipSigning);
-        Optional.ofNullable(staticLibs).ifPresent(config::setJavaStaticLibs);
-        Optional.ofNullable(staticJavaFXSDK).ifPresent(config::setJavaFXStaticSDK);
-        if (reflectionList != null && !reflectionList.trim().isEmpty()) {
-            config.setReflectionList(Arrays.asList(reflectionList.split(",")));
-        }
-        if (jniList != null && !jniList.trim().isEmpty()) {
-            config.setJniList(Arrays.asList(jniList.split(",")));
-        }
-        if (bundlesList != null && !bundlesList.trim().isEmpty()) {
-            config.setBundlesList(Arrays.asList(bundlesList.split(",")));
-        }
+        config.setJavaStaticLibs(staticLibs);
+        config.setJavaFXStaticSDK(staticJavaFXSDK);
```

Comment:
should not be set by a developer

Reply:
Which one? Whats the default behavior for this?

Reply:
The InternalConfiguration